cogs/roles.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) idoneam (2016-2019)
+#
+# This file is part of Canary
+#
+# Canary is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Canary is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Canary. If not, see <https://www.gnu.org/licenses/>.
+
+# discord-py requirements
+import discord
+from discord import utils
+from discord.ext import commands
+
+import codecs
+import configparser
+from .utils.paginator import Pages
+
+
+class Roles(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.roles = self.bot.config.roles
+
+    async def toggle_role(self, ctx, transaction, requested_role, category):
+        """
+        Assigns a single role to a user with no checks from a category of roles
+        """
+        member = ctx.message.author
+        roles = self.roles[category]
+
+        # normalize user input to title case
+        requested_role = requested_role.title()
+        role = discord.utils.get(ctx.guild.roles, name=requested_role)
+
+        if (requested_role not in roles):
+            await ctx.send("Invalid Role")
+            return
+
+        if (transaction == "add"):
+            await member.add_roles(role, reason="Self Requested")
+        elif (transaction == "remove"):
+            await member.remove_roles(role, reason="Self Requested")
+        else:
+            await ctx.send("Must `add` or `remove` a role")
+
+    async def toggle_unique_role(self, ctx, transaction, requested_role,
+                                 category):
+        """
+        Assigs a single role to a user with no checks from a category of roles
+        """
+        member = ctx.message.author
+        roles = self.roles[category]
+
+        # normalize user input to title case
+        requested_role = requested_role.title()
+        role = discord.utils.get(ctx.guild.roles, name=requested_role)
+
+        if (requested_role not in roles):
+            await ctx.send("Invalid Role")
+            return
+
+        if (transaction == "add"):
+            # Allow only a single faculty role per user.
```

Comment:
This comment could be made a bit more general since this function isn't going to be called just for faculty role assignment.